src/test/java/org/sagebionetworks/template/Ec2ClientImplTest.java

```diff
@@ -0,0 +1,168 @@
+		// Call under test
+		List<String> azsForInstanceType = ec2Client.getAvailabilityZonesForInstanceType(INSTANCE_TYPE);
+		assertNotNull(azsForInstanceType);
+		assertEquals(4, azsForInstanceType.size());
+		DescribeInstanceTypeOfferingsRequest r = describeInstanceOfferingsRequestCaptor.getValue();
+		assertEquals(1, r.getFilters().size());
+		Filter f = r.getFilters().get(0);
+		assertEquals("instance-type", f.getName());
+		assertEquals(1, f.getValues().size());
+		assertEquals(INSTANCE_TYPE, f.getValues().get(0));
+	}
+
+	@Test
+	void getAvailableSubnetsForInstanceType() {
+		when(mockEC2.describeSubnets(any(DescribeSubnetsRequest.class))).thenReturn(mockDescribeSubnetsResult);
+		when(mockEC2.describeInstanceTypeOfferings(any(DescribeInstanceTypeOfferingsRequest.class))).thenReturn(mockDescribeInstanceTyepOfferingsResult);
+		List<Subnet> expectedSubnets = generateSubnets(6);
+		when(mockDescribeSubnetsResult.getSubnets()).thenReturn(expectedSubnets);
+		List<InstanceTypeOffering> expectedOfferings = generateInstanceOfferings(INSTANCE_TYPE);
+		when(mockDescribeInstanceTyepOfferingsResult.getInstanceTypeOfferings()).thenReturn(expectedOfferings);
+		List<String> subnets = Arrays.asList("subnet1", "subnet2", "subnet3", "subnet4", "subnet5", "subnet6");
+		// Call under test
+		List<String> availableSubnets = ec2Client.getAvailableSubnetsForInstanceType(INSTANCE_TYPE, subnets);
+		assertNotNull(availableSubnets);
+		List<String> expectedAvailableSubnets = Arrays.asList("subnet1", "subnet2", "subnet4", "subnet5");
+		assertEquals(expectedAvailableSubnets, availableSubnets);
+	}
+
+	@Test
+	void getAvailableSubnetsForInstanceTypeTooSmall() {
+		when(mockEC2.describeSubnets(any(DescribeSubnetsRequest.class))).thenReturn(mockDescribeSubnetsResult);
+		when(mockEC2.describeInstanceTypeOfferings(any(DescribeInstanceTypeOfferingsRequest.class))).thenReturn(mockDescribeInstanceTyepOfferingsResult);
+		List<Subnet> expectedSubnets = generateSubnets(6);
+		when(mockDescribeSubnetsResult.getSubnets()).thenReturn(expectedSubnets);
+		List<InstanceTypeOffering> expectedOfferings = generateInstanceOfferingsTooSmall(INSTANCE_TYPE);
+		when(mockDescribeInstanceTyepOfferingsResult.getInstanceTypeOfferings()).thenReturn(expectedOfferings);
+		List<String> subnets = Arrays.asList("subnet1", "subnet2", "subnet3", "subnet4", "subnet5", "subnet6");
+		// Call under test
+		IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> {ec2Client.getAvailableSubnetsForInstanceType(INSTANCE_TYPE, subnets);});
+		assertEquals(String.format("Could not find 2 available subnets for type %s in %s ", INSTANCE_TYPE, subnets), e.getMessage());
+
+	}
+
+	/**
+	 * This maps subnet1 to us-east-1a, subnet2 to us-east-1b etc.
+	 * @param numSubnets
+	 * @return
+	 */
+	private List<Subnet> generateSubnets(int numSubnets) {
+		List<Subnet> l = new ArrayList<>(numSubnets);
```

Comment:
You could use an IntStream:
IntStream.of(5).boxed().map( i -> new Subnet().withSubnetId(String.format("subnet%d", i+1)));